td-reproducible-tool/readme.md

```diff
@@ -0,0 +1,61 @@
+# td-reproducible-tool
+
+## Background
+
+Elements break reproducible build (x86_64-unknown-uefi + target x86_64-unknown-none) <br>
+ <br>
+```
+1. Debug related sections (PE + ELF) 
+2. Code base path like \home\USERNAME\rust-td-payload\xxx\xxx.rs (PE + ELF) 
+3. CARGO_HOME like \home\USERNAME\.cargo\xxx\xxx.rs (PE + ELF) 
+4. RUSTUP_HOME like \home\USERNAME\.rustup\xxx\xxx.rs (PE + ELF) 
+5. TimeDateStamp field in COFF file header (PE only) 
+```
+
+Solution: <br>
+

```

Comment:
Could 2, 3, 4 also be solved by using build container to normalize the build env?

Reply:
It is based upon the definition of "reproducible build".

In our definition, "reproducible build" means
1) Build with same code and version
2) Build with same compiler and version
3) Build at different time.
4) Build at different directory/path <==

Reply:
The final design: 
1 and 2 are handled by build option.
3 and 4 are handled by post-build tool as an option.
5. Timestamp is always cleared.